iotedgedev/decorators.py

```diff
@@ -0,0 +1,75 @@
+import hashlib
+import sys
+from functools import wraps
+
+
+def with_telemetry(func):
+    @wraps(func)
+    def _wrapper(*args, **kwargs):
+        from . import telemetry
+        from .telemetryconfig import TelemetryConfig
+
+        config = TelemetryConfig()
+        config.check_firsttime()
+        params = parse_params(*args, **kwargs)
+        telemetry.start(func.__name__, params)
+        try:
+            value = func(*args, **kwargs)
+            telemetry.success()
+            telemetry.flush()
+            return value
+        except Exception as e:
+            from .output import Output
+            Output().error('Error: {0}'.format(str(e)))
+            telemetry.fail(str(e), 'Command failed')
+            telemetry.flush()
+            sys.exit(1)
+
+    return _wrapper
+
+
+def suppress_all_exceptions(fallback_return=None):
+    """We need to suppress exceptions for some internal functions such as those related to telemetry.
+    They should not be visible to users.
+    """
+    def _decorator(func):
+        @wraps(func)
+        def _wrapped_func(*args, **kwargs):
+            try:
+                return func(*args, **kwargs)
+            except Exception:
+                if fallback_return:
+                    return fallback_return
+                else:
+                    pass
+
+        return _wrapped_func
+
+    return _decorator
+
+
+@suppress_all_exceptions()
+def parse_params(*args, **kwargs):
```

Comment:
Couple things here. First None is portrayed correctly with format ('{}={}'.format('key', None) >>> 'key=None') but its fine.  Also what if value is an empty string like ""  or just contains spaces like " "? Both of these will evaluate as key!=None is that OK?